xarray/core/dataarray.py

```diff
@@ -534,10 +534,14 @@ def _replace_maybe_drop_dims(
         variable: Variable,
         name: Hashable | None | Default = _default,
     ) -> Self:
-        if variable.dims == self.dims and variable.shape == self.shape:
+        dims_are_equal = set(variable.dims) == set(self.dims)
+        size_is_equal = (
+            self.sizes[dim] == size for dim, size in variable.sizes.items()
+        )
+        if dims_are_equal and all(size_is_equal):
```

Comment:
```suggestion
        variable.sizes == dims.sizes
```
?

(Seems to not check order)

Reply:
🤦🏾‍♂️